src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPageCount, setPerPageCount] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const totalItems = 42;
+
+  const firstItemOnPage = (currentPage - 1) * perPageCount;
+
+  const lastItemOnPage = firstItemOnPage + perPageCount;
+
+  const amountOfItems = getNumbers(1, 42).filter(
```

Comment:
u can get only the items u need on a current page. u don't get everything at once